src/main/java/com/eva/model/person/staff/leave/Leave.java

```diff
@@ -0,0 +1,87 @@
+package com.eva.model.person.staff.leave;
+
+import static com.eva.commons.util.AppUtil.checkArgument;
+import static com.eva.commons.util.DateUtil.DATE_TIME_FORMATTER;
+import static com.eva.commons.util.DateUtil.isValidDate;
+import static java.util.Objects.requireNonNull;
+
+import java.time.LocalDate;
+import java.time.Period;
+import java.util.Objects;
+
+
+/**
+ * Represents a Tag in the address book.
+ * Guarantees: immutable; name is valid as declared in {@link #isValidLeaveDate(String)}
+ */
+public class Leave {
+
+    public static final String MESSAGE_CONSTRAINTS = "Leave date should be in the format dd/mm/yyyy";
+    private static final String SINGLE_DAY_LEAVE = "Leave taken on %s";
+    private static final String MULTIPLE_DAY_LEAVE = "Leave taken from %s to %s";
+
+    private final LocalDate from;
+    private final LocalDate to;
+    private final Period period;
+
+    /**
+     * Constructs a {@code Leave}.
+     *
+     * @param date A valid leave start date.
+     */
+    public Leave(String date) {
+        requireNonNull(date);
+        checkArgument(isValidLeaveDate(date), MESSAGE_CONSTRAINTS);
+        this.from = LocalDate.parse(date, DATE_TIME_FORMATTER);
+        this.to = from;
+        period = Period.ofDays(1);
+    }
+
+    /**
+     * Constructs a {@code Leave}.
+     *
+     * @param from A valid leave start date.
+     * @param to   A valid leave end date.
+     */
+    public Leave(String from, String to) {
```

Comment:
Was thinking if this can be improved to maintain the same level of abstraction as before, by constructing the Leave object with ` public Leave(LocalDate from, LocalDate to) {` because in other cases for example, 
Add Person, the text input goes from Logic, -->  AddressBookParser --> AddCommandParser and this is where the Name, Email, Address Objects are made, then these objects are passed into AddCommand and is then passed into Model to add the person. 
I think this change does not make a difference logic wise, but helps make the level of abstraction more standard. But up to you to take it up.

Reply:
I believe a better comparison would be between Leave and Address or Tag! Leave should handled the same way as how an attribute of the person is handled instead of being handled the same way as the Person itself.

Reply:
Alright sure, I agree with your reasoning, Thank you! Great job :)